coordinator/src/node/rollover.rs

```diff
@@ -215,6 +252,44 @@ impl Node {
             contract.get_temporary_id(),
         )
     }
+
+    pub fn rollback_channel_if_needed(
+        &self,
+        connection: &mut PgConnection,
+        signed_channel: SignedChannel,
+    ) -> Result<(bool, Option<ContractId>)> {
+        if is_channel_in_intermediate_state(&signed_channel) {
+            let trader_id = signed_channel.counter_party;
+            tracing::warn!(%trader_id, "Found signed channel contract in an intermediate state. Trying to rollback channel to the last stable state!");
+            self.inner
+                .rollback_channel(&signed_channel)
+                .map_err(|e| anyhow!("{e:#}"))?;
+
+            let contract = self
+                .inner
+                .get_contract_by_dlc_channel_id(&signed_channel.channel_id)?;
+            db::positions::Position::set_position_to_open(
```

Comment:
Why do you set the position to open, shouldn't it be open already?

Reply:
in case we are in a stuck intermediate rollover state the position is in state `Rollover`. For a proper rollback we have to set the position back to `Open`.